### PR TITLE
Update docs with interactive examples

### LINKED ISSUE #133: Use the new PyVista sphinx directives for interactive plots in the documentation?
It'd be great to make the documentation examples interactive! I'll see if I can make some time to contribute this while on winter holiday

### PR DESCRIPTION
**Pull request by banesullivan**
I think this is all that is needed to resolve #133 -- at least its working locally for me. I suspect we might need to add other dependencies though? Let's see if CI is happy

I also changed the API docs so that that the long function names weren't appearing in the navbar

### CODE REVIEW

**Comment by banesullivan** [comment]
@MatthewFlamm, one suggestion I have is to turn on edge visibility in the global theme for the docs here. I think it helps alot to show the cell edges and where the sampled points lie with respect to the cells and their resolution

File: doc/conf.py [hunk 1]
```diff
@@ -42,6 +45,7 @@
 pyvista.global_theme.font.label_size = 22
 pyvista.global_theme.font.title_size = 22
 pyvista.global_theme.return_cpos = False
+pyvista.global_theme.show_edges = True
```
Comment (banesullivan) [inline doc/conf.py:48]: For https://github.com/MatthewFlamm/pyransame/pull/136#issuecomment-1868364974

If you're not a fan of this, feel free to remove:

```suggestion
```

Comment (banesullivan) [reply doc/conf.py:48]: Wait nevermind, this is not propagating to the plot directive 😕

**Comment by MatthewFlamm** [comment]
Thanks!  This still isn't working for me, at least on a codespaces container, but maybe there is some additional port forwarding I need to do.  I won't have a local machine for testing for awhile. I did:

```py
pip install -r requirements-doc.txt
cd doc
make html
make serve-html
```

File: doc/conf.py [hunk 2]
```diff
@@ -18,6 +18,9 @@
     "numpydoc",
     "sphinx.ext.autosummary",
     "pyvista.ext.plot_directive",
+    "pyvista.ext.plot_directive",
```
Comment (MatthewFlamm) [inline doc/conf.py:21]:
```suggestion
```

File: requirements-doc.txt [hunk 3]
```diff
@@ -1,3 +1,8 @@
 numpydoc==1.6.0
 pydata-sphinx-theme==0.14.4
+pyvista==0.43.1
 sphinx==7.2.6
+sphinx-design==0.5.0
+trame==3.3.0
```
Comment (MatthewFlamm) [inline requirements-doc.txt:6]:
```suggestion
sphinx-toolbox==3.5.0
trame==3.3.0
```

**Comment by MatthewFlamm** [comment]
I had tested that installing all the PyVista requirements for documentation worked.  I tried uninstalling those and then added `sphinx-toolbox` back and it worked.  But, there must be some sub-dependency that is also required as just adding `sphinx-toolbox` does not work for me.

I can confirm that serving the html from codespaces works as expected.

**Comment by MatthewFlamm** [comment]
Some dependency, or dependencies, of `jupyterlab` are required.  I don't think `jupyterlab` itself is required as I had uninstalled it originally and it had worked.  But it is probably easiest just to keep that top level dependency for now.

**Review by MatthewFlamm — APPROVED** [review]
LGTM.  Thanks!

**Comment by MatthewFlamm** [comment]
Coming back to this, I pushed the changes to GitHub Pages, so this is now live.  It looks great.  Thanks for your help on this again!  I also switched over to using `sphinx-gallery` for most of the heavy lifting as the ability to mix plots and code output with preserving state across blocks is really useful.

I had to revamp the examples to use glyphs for spheres as `pv.add_points(mesh, render_points_as_spheres=True)` leads to spheres that stay constant size with respect to the viewport rather than the mesh.  The interactive plots didn't work well in that case.

**Comment by banesullivan** [comment]
Thanks for pointing that out! I too have come back to this PR a few times to see how its done to repeat in work like https://github.com/bjlittle/geovista/pull/632. We really ought to better document this in PyVista

**Pull request merged**